### PR TITLE
Update README to highlight restore database requirements

### PR DESCRIPTION
**Pull request by eiri**
## Checklist

- [x] Tick to sign-off your agreement to the [Developer Certificate of Origin (DCO) 1.1](../blob/master/DCO1.1.txt)
- [ ] Added tests for code changes _or_ test/build only changes
- [ ] Updated the change log file (`CHANGES.md`|`CHANGELOG.md`) _or_ test/build only changes
- [ ] Completed the PR template below:

## Description

Update README to emphasize that target database for restore should be new and empty database.

Ref: #367

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -122,7 +122,7 @@ couchbackup --db animaldb --log animaldb.log --resume true --output animaldb.txt
 
 ## Restore
 
-Now we have our backup text file, we can restore it to an existing database using the `couchrestore`:
+Now we have our backup text file, we can restore it to a new, empty, existing database using the `couchrestore`:
```
Comment (emlaver) [inline README.md:125]:
```suggestion
Now that we have our backup text file, we can restore it to a new, empty, existing database using the `couchrestore`:
```

File: README.md [hunk 2]
```diff
@@ -319,8 +319,10 @@ couchbackup.backup(
 ### Restore
 
 The `restore` function takes a readable stream containing the data emitted
-by the `backup` function. It uploads that to a Cloudant database, which
-should be a **new** database.
+by the `backup` function and uploads that to a Cloudant database
```
Comment (emlaver) [inline README.md:322]:
```suggestion
by the `backup` function and uploads that to a Cloudant database.
```

**Review by emlaver — APPROVED** [review]
looks good!

File: README.md [hunk 3]
```diff
@@ -319,8 +319,10 @@ couchbackup.backup(
 ### Restore
 
 The `restore` function takes a readable stream containing the data emitted
-by the `backup` function. It uploads that to a Cloudant database, which
-should be a **new** database.
+by the `backup` function and uploads that to a Cloudant database.
+
+_Note:_ A target database should be a **new and empty** database to avoid
+possible documents' update conflicts.
```
Comment (ricellis) [inline README.md:325]: I think this a _must_ (particularly if we're doing #376) and I think as a requirement of the restore then the _why_ isn't adding a lot of value.
```suggestion
_Note:_ A target database must be a **new and empty** database.
```

**Review by ricellis — APPROVED** [review]
1 nit, but I'm +1 anyway

**Pull request merged**